flax/cursor.py

```diff
@@ -49,97 +51,138 @@ def is_named_tuple(obj):
   )
 
 
-def _get_changes(path, obj, update_fn):
-  """Helper function for ``Cursor.apply_update``. Returns a generator of
-  Tuple[Tuple[Union[str, int], Any], ...], where the first element is a
-  tuple key path where the change was applied from the ``update_fn``, and
-  the second element is the newly modified value. If the generator is
-  non-empty, then the tuple key path will always be non-empty as well."""
+def _traverse_tree(path, obj, *, update_fn=None, cond_fn=None):
+  """Helper function for ``Cursor.apply_update`` and ``Cursor.find_all``.
+  Exactly one of ``update_fn`` and ``cond_fn`` must be not None.
+
+  - If ``update_fn`` is not None, then ``Cursor.apply_update`` is calling
+    this function and ``_traverse_tree`` will return a generator where
+    each generated element is of type Tuple[Tuple[Union[str, int], AccessType], Any].
+    The first element is a tuple of the key path and access type where the
+    change was applied from the ``update_fn``, and the second element is
+    the newly modified value. If the generator is non-empty, then the
+    tuple key path will always be non-empty as well.
+  - If ``cond_fn`` is not None, then ``Cursor.find_all`` is calling this
+    function and ``_traverse_tree`` will return a generator where each
+    generated element is of type Tuple[Union[str, int], AccessType]. The
+    tuple contains the key path and access type where the object was found
+    that fulfilled the conditions of the ``cond_fn``.
+  """
+  if (update_fn and cond_fn) or (update_fn is None and cond_fn is None):
```

Comment:
```suggestion
  if update_fn ^ cond_fn: # xor
```

Reply:
I think `^` only works with boolean operands, so I could do something like this:
```
if not (bool(update_fn) ^ bool(cond_fn)):
```
What do you think?

Reply:
Yeah, I sorry missed. This looks good!